Support multiple weekdays and ordinal days

jbondia opened the pull request:
Allowing any weekday, month, ordinal days comma separated values.

adamchainz commented:
1. Please add tests! I see you have a TODO comment for that in your second commit message
2. Remove all `pylint` comments - we don't lint this repository with pylint, if you do personally it's your responsibility to disable it whilst editing

jbondia commented:
1. Yep, sorry about that.
2. Understood

I will do it after work.

Thanks

adamchainz commented:
Also:
3. Add yourself to AUTHORS.rst as a contributor! Alphabetical order by first name please.
4. Add a line to HISTORY.rst underneath 'New release notes here'.

jbondia commented:
If I am not wrong:
1. Deleted pylint comments.
2. Added tests for weekdays and ordinal days with comma separated values (months already supported).
3. Added as contributor with first name alphabetical order.
4. Added history line describing the changes.

Regards,

adamchainz commented on HISTORY.rst:
```diff
@@ -8,6 +8,11 @@ Pending Release
 
 * New release notes here
 
+1.0.2 (2016-04-27)
```
please don't add release header, i'll do that on release

adamchainz commented on pretty_cron/api.py:
```diff
@@ -65,13 +72,18 @@ def _pretty_date(month_day, month, week_day):
             )
 
         pretty_date = " and ".join(
+            # pylint: disable=W0141
```
pylint comment still here

adamchainz commented on pretty_cron/api.py:
```diff
@@ -1,10 +1,15 @@
-    for i, piece in enumerate(expression.split(" ")):
-        if i == 3 and ',' in piece:  # support comma-separated values for month
+    for piece in expression.split(" "):
+        # support comma-separated values for ordinal days, weekdays and months
```
if this only supports days, weekdays, and months, shouldn't the conditional include, like before, `if i in (1, 2, 3)` ? otherwise you're allowing `,` in hours

jbondia replied:
I guess you are right. Sorry about this.
What about 'i != 0 and [...]' instead?

adamchainz commented on pretty_cron/api.py:
```diff
@@ -65,13 +72,18 @@ def _pretty_date(month_day, month, week_day):
 
 def _human_month(month):
+    """
+    Cron-tab to month
```
vacant docstring, please remove (it says no more than the function name)

adamchainz commented on pretty_cron/api.py:
```diff
@@ -106,7 +121,17 @@ def _human_list(listy):
 def _human_week_day(day):
-    return _WEEKDAYS[day]
+    """
+    Cron-tab to weekday
```
ditto, vacant docstring, please remove (just because pylint tells you to do something...)

jbondia replied:
As before, you are right. Do you prefer me to remove all the docstrings or write them in the right way explaining parameters and return stuff?

adamchainz commented on pretty_cron/api.py:
```diff
@@ -117,15 +142,29 @@ def _human_week_day(day):
-    if 10 <= (n % 100) < 20:
-        suffix = 'th'
+    """
+    Cron-tab to numeric day
```
ditto plz remove

adamchainz commented on pretty_cron/api.py:
```diff
@@ -117,15 +142,29 @@ def _human_week_day(day):
 
 def _pretty_time(minute, hour):
+    """
+    Cron-tab to time
```
ditto

adamchainz commented on pretty_cron/api.py:
```diff
@@ -117,15 +142,29 @@ def _human_week_day(day):
+    """
+    Cron-tab to numeric day
+    """
+    if isinstance(n, tuple):
```
this is simpler as

``` py
if not isinstance(n, tuple):
    n = (n,)
```

jbondia replied:
OMG, yeah. Thanks. I guess we also could apply this change at _human_month and _human_week_day?

adamchainz commented on .gitignore:
```diff
@@ -3,6 +3,10 @@
 # C extensions
 *.so
 
+# vim temporal files
```
please don't do this and introduce unnecessary change, if your editor generates extra temporary files then ignore them with a global git ignore like https://help.github.com/articles/ignoring-files/

jbondia replied:
Sorry Adam, and thank you for your patience. I am learning quite a lot of things today..

adamchainz replied:
No worries, glad to help! Thanks for working with me in my push for perfection 👍

adamchainz commented:
Merged in edd42a5df913b97649b72cacc1e5a29505a04b02 , thanks!

jbondia commented:
Thanks a lot for your help Adam

adamchainz commented:
Sorry, it was 0e5d729e0c0d50715514de502be669ed9bdb2267 that merged it, commit SHA got changed when I rebased it onto master

adamchainz commented:
If you were waiting for PyPI release, I just did it as 1.0.2: https://pypi.python.org/pypi/pretty-cron

jbondia commented:
Thanks a lot, Adam!

Julio Bondia Barceló

On Tue, May 3, 2016 at 6:05 PM, Adam Chainz notifications@github.com
wrote:

> If you were waiting for PyPI release, I just did it as 1.0.2:
> https://pypi.python.org/pypi/pretty-cron
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/adamchainz/pretty-cron/pull/15#issuecomment-216597045
